Code/Framework/AzCore/AzCore/Debug/TraceReflection.cpp

```diff
@@ -48,63 +94,197 @@ namespace AZ
             bool OnPrintf(const char* window, const char* message) override;
             bool OnOutput(const char* window, const char* message) override;
 
+            // AZ::TickBus::Handler overrides ...
+            void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+            int GetTickOrder() override;
+
         private:
-            template<class R, class... Args>
-            R CallResultReturn(const R& defaultReturnValue, int index, Args&&... args) const
+            void QueueMessageCall(AZStd::function<void()> messageCall);
+            void FlushMessageCalls();
+
+            AZStd::list<AZStd::function<void()>> m_messageCalls;
```

Comment:
Is there a reason to prefer list instead of vector here?

Reply:
Since it's not clear the size this array will have, I didn't use a vector to avoid reallocating the memory and moving all the elements with insertions. In addition, a list provides fast insertion, which is great to keep the mutex locked the less time possible.